src/aiida/__init__.py

```diff
@@ -35,6 +36,8 @@
 )
 __paper_short__ = 'S. P. Huber et al., Scientific Data 7, 300 (2020).'
 
+# Initialize the configuration directory settings
+AiiDAConfigPathResolver.set_configuration_directory()
```

Comment:
I believe this is correct, if so this is way better. Previously, it was always a mystery to me where my `~/.aiida` folder get created.

Reply:
This unfortunately can't be here, we don't want to initialize the configuration when you do `import aiida`. 

Configuration was very intentionally done lazily, so that it doesn't run when you do tab completion in verdi cli. 

I agree that the current state was also not clear, let's discuss this in person maybe.

Reply:
My quick thought is that this should be called at the end of `settings.py` as before, I currently don't see another solution.

Reply:
I have explicitly made sure that the settings module is only imported when it is really needed. Thats why for example its import is inlined in `manage/configuration/__init__.py`, 

I think (happy to be proven wrong!) that if the singleton class is to be working robustly, it needs to be populated with defaults as soon as it is created, and the class is created when the module is first imported.